File: openfeed-backend/openfeed/db/global_settings.py
from openfeed.db.client import Client
from openfeed.database_models import PublicGlobalSettings


def get_global_settings(db: Client) -> PublicGlobalSettings:
    data = (
        db.table("global_settings")
        .select("*")
        .eq("singleton", True)
        .single()
        .execute()
        .data
    )
    return PublicGlobalSettings.model_validate(data)

File: openfeed-frontend/lib/supabase/queries/global_settings.ts
import { SupabaseClient } from "@supabase/supabase-js";
import { Database } from "@/lib/supabase/supabase.types";

export type GlobalSettings =
  Database["public"]["Tables"]["global_settings"]["Row"];

export async function getGlobalSettings(
  supabase: SupabaseClient<Database>,
): Promise<GlobalSettings> {
  const { data, error } = await supabase
    .from("global_settings")
    .select("*")
    .eq("singleton", true)
    .single();

  if (error) throw new Error(error.message);

  return data;
}
